lib/Executor/ScriptExecutor.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace PhpBench\Executor;
+
+use PhpBench\Remote\Exception\ScriptErrorException;
+use RuntimeException;
+use Symfony\Component\Process\Process;
+use function uniqid;
+
+final class ScriptExecutor
+{
+    /**
+     * @var string|null
+     */
+    private $scriptPath;
+
+    /**
+     * @var bool
+     */
+    private $scriptRemove;
+
+    /**
+     * @var PhpProcessFactory
+     */
+    private $factory;
+
+    public function __construct(
+        PhpProcessFactory $factory,
+        ?string $scriptPath,
+        bool $scriptRemove
+    ) {
+        $this->scriptPath = $scriptPath;
+        $this->factory = $factory;
+        $this->scriptRemove = $scriptRemove;
+    }
+
+    /**
+     * @return array<string,mixed>
+     */
+    public function execute(/** include the run ID here **/string $script): array
```

Comment:
the idea was to include the run ID so that it can be used to namespace the file rather than use a `uniqid`.